### PR TITLE
Remove compat build

### PR DESCRIPTION
**Pull request by hasanakg**
Bug: T351812

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -59,7 +59,6 @@
     "dependencies": {
         "@inertiajs/inertia": "^0.11.0",
         "@inertiajs/inertia-vue3": "^0.6.0",
-        "@vue/compat": "^3.3.8",
         "@wmde/wikit-tokens": "^2.1.0-alpha.15",
         "@wmde/wikit-vue-components": "^2.1.0-alpha.16",
```
Comment (chukarave) [inline package.json:63]: I think we can remove lines 62 and 63 too

File: resources/js/Components/LoadingOverlay.vue [hunk 2]
```diff
@@ -14,8 +14,8 @@ import { ref } from 'vue';
 import { CdxProgressBar } from "@wikimedia/codex";
 
 const props = withDefaults(defineProps<{
-	delay: number
-	visible: boolean
+	delay?: number
+	visible?: boolean
```
Comment (chukarave) [inline resources/js/Components/LoadingOverlay.vue:18]: Isn't making these optional kind of risky? If we somehow don't pass `visible` the `show()` method breaks. Is there a way around this that leaves them required?

Comment (hasanakg) [reply resources/js/Components/LoadingOverlay.vue:18]: They have default values, therefore they won't be `null` or `undefined`

File: package.json (hunk 1 above)
Comment (hasanakg) [reply package.json:63]: I think removing `wikit` is not in the scope of this ticket, that's why I left these lines.

File: jest.config.js [hunk 3]
```diff
@@ -22,8 +22,6 @@ module.exports = {
     // "import Counter from '@/Counter.vue'"
     // (no need for a full path)
     "moduleNameMapper": {
-        "^vue$": "@vue/compat",
-        '^@vue/composition-api$': '@vue/compat',
         '^@wmde/wikit-vue-components$':
             '@wmde/wikit-vue-components/dist/wikit-vue-components-vue3compat.common.js',
         '^wikit-dist(.*)$': "<rootDir>/node_modules/@wmde/wikit-vue-components/dist$1",
```
[the remark this answers is not in the document]
Comment (guergana) [reply jest.config.js:27]: > I think removing `wikit` is not in the scope of this ticket, that's why I left these lines.

@hasanakg You are right that this is not the scope of this ticket but i think there is no ticket to remove wikit yet, we should do a cleanup afterwards.

File: resources/js/Components/LoadingOverlay.vue (hunk 2 above)
Comment (guergana) [reply resources/js/Components/LoadingOverlay.vue:18]: > They have default values, therefore they won't be `null` or `undefined`

I also had this issue, i was getting a TS warning or error because of this at some point but i dont see it anymore and they do have defaults, but my question is... is this part of the scope of this ticket? :grimacing:

Comment (hasanakg) [reply resources/js/Components/LoadingOverlay.vue:18]: Valid question :D Vue documentation says that you should remove all the warnings before removing the compat build, I took it as a sign :D

Comment (guergana) [reply resources/js/Components/LoadingOverlay.vue:18]: > Valid question :D Vue documentation says that you should remove all the warnings before removing the compat build, I took it as a sign :D

ah ok, so you got the warning.. i also saw it. yes, we shouldn't merge code with warnings :)

**Comment by hasanakg** [comment]
> The `package-lock.json` file is missing from the changes.

Fixed now.

**Review by guergana — CHANGES_REQUESTED** [review]
As part of this task we should remove any references to the compat build and extra imports / hacks we used, like these two:

```
// Run it with compat mode
// https://v3-migration.vuejs.org/breaking-changes/v-model.html
CdxTextArea.compatConfig = {
    ...CdxTextArea.compatConfig,
    COMPONENT_V_MODEL: false,
};
```
in https://github.com/wmde/wikidata-mismatch-finder/blob/development/resources/js/Components/ItemIdSearchTextarea.vue#L33

and 

```
// Run it with compat mode
// https://v3-migration.vuejs.org/breaking-changes/v-model.html
CdxCheckbox.compatConfig = {
    ...CdxCheckbox.compatConfig,
    COMPONENT_V_MODEL: false,
};
```
in https://github.com/wmde/wikidata-mismatch-finder/blob/development/resources/js/Pages/Results.vue#L202

File: resources/js/Components/LoadingOverlay.vue (hunk 2 above)
Comment (guergana) [reply resources/js/Components/LoadingOverlay.vue:18]: > Isn't making these optional kind of risky? If we somehow don't pass `visible` the `show()` method breaks. Is there a way around this that leaves them required?

@chukarave the only instance of the loading overlay we are using in the app is not passing the props and this is the reason we are getting the warnings `<loading-overlay ref="overlayRef" />` https://github.com/wmde/wikidata-mismatch-finder/blob/development/resources/js/Pages/Results.vue#L3 so yeah, the right call is to make them optional, since they are not being passed anyway

File: tests/Vue/Components/MismatchRow.spec.js [hunk 4]
```diff
@@ -275,7 +275,7 @@ describe('MismatchesRow.vue', () => {
 
         await wrapper.find('.full-description-button').trigger('click');
 
-        const dialog = wrapper.find('.full-description-dialog .cdx-dialog');
```
Comment (guergana) [inline tests/Vue/Components/MismatchRow.spec.js:278]: everything is working flawlessly, but how is this related to the compat build??? :thinking:

Comment (hasanakg) [reply tests/Vue/Components/MismatchRow.spec.js:278]: TBH, I have no idea. The old one works with compat build, but needed to update to the new one to make it work without compat build.

**Review by guergana — APPROVED** [review]
Works flawlessly, thanks

File: resources/js/Components/LoadingOverlay.vue (hunk 2 above)
Comment (chukarave) [reply resources/js/Components/LoadingOverlay.vue:18]: Right those were exactly the warnings we were getting. Nice to have a clean console :raised_hands: well done @hasanakg
